### ISSUE TITLE
Error with json with several levels in several lines

### ISSUE THREAD

**Issue by JulioJu** [sharat87/roast.vim#5]
**Error**

The following roast file give the error presented below when we type <ENTER> in the second GET URL. No any request is sent and the right pan of Vim is opened on a blank window. 

```
GET https://httpbin.org/get name=Sherlock

POST https://httpbin.org/forms/post << END
{{
    "essai": 1,
    "toto": [
        {{"toto": "toto"}}
    ],
    "essai2": 2
}}
END

# Can't work
GET https://httpbin.org/get name=Sherlock

```

```
Error detected while processing function roast#run:
line    1:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/julioprayer/.vim/plugged/roast.vim/python3/roast.py", line 40, in run
    highlight_line_text('RoastCurrentSuccess' if response.ok else 'RoastCurrentFailure')
  File "/home/julioprayer/.vim/plugged/roast.vim/python3/roast.py", line 98, in highlight_line_text
    vim.current.buffer.vars['_roast_match_id'] = vim.eval(f"matchadd('{group}', '\\V{vim.current.line}')")
vim.error: Vim:E28: No such highlight group name: RoastCurrentFailure

```
**Note**
The following works perfectly in the second GET URL . The right pan of Vim is opened with the correct JSON response. 

Note that `toto` is in one line.

```
# Works
GET https://httpbin.org/get name=Sherlock

POST https://httpbin.org/forms/post << END
{{
    "essai": 1,
    "toto": [ {{"toto": "toto"}} ],
    "essai2": 2
}}
END

# Works
GET https://httpbin.org/get name=Sherlock
```

**Comment by sharat87** [comment]
I see two problems here. Firstly, the error about `RoastCurrentFailure`, which should be fixed in #4.

The second is the missing heredoc termination. The line with "END" after all the JSON is missing in your first example, and is present in the second example. Without this line, Roast wouldn't know where your POST body ends.

Please re-open if you have further questions.

**Comment by JulioJu** [comment]
@sharat87 thanks for your answer,

So sorry for my mystake. I've corrected the error above and added « END » at the line 12.  But doesn't resolve the problem.

If you test the example above, the second GET at the last line doesn't work. When you press ENTER  on the second GET you have. 

```
=== Response Headers ===
Content-Length: 0
Connection: Close


=== Request Headers ===
Essai: 1,
Toto: [
{{Toto: toto}
Essai2: 2
```

I can't open again this issue, I have not the button to pen it again.

**Comment by JulioJu** [comment]
@sharat87 thanks for your answer,

So sorry for my mystake. I've corrected the error above and added « END » at the line 12.  But doesn't resolve the problem.

If you test the example above, the second GET at the last line doesn't work. When you press ENTER  on the second GET you have. 

```
=== Response Headers ===
Content-Length: 0
Connection: Close


=== Request Headers ===
Essai: 1,
Toto: [
{{Toto: toto}
Essai2: 2
```

I can't open again this issue, I have not the button to pen it again.

**Comment by sharat87** [comment]
Hey Julio, I didn't realize the second example failed even with the `END` line. It appears this is a bug.

For the record, I've prepared a minimal test case that reproduces this problem. I'm working towards a solution to this currently.

```
POST https://httpbin.org/post << END
{{"toto": "toto"}}
END

# Can't work
GET https://httpbin.org/get name=Sherlock
```

The problem is that Roast is treating line 2 as a header definition, which is incorrect.

**Comment by sharat87** [comment]
Should be fixed in 7d3aa60. Added a test so it doesn't show up again in the future.

**Comment by JulioJu** [comment]
@sharat87 thanks a lot ;-)
